src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -281,6 +297,76 @@ public bool IsSolutionOpen
             }
         }
 
+        public async Task<Tuple<bool, bool>> CheckSolutionUIPreConditionsAsync()
+        {
+            // Do NOT initialize VSSolutionManager through this API (by calling EnsureInitializeAsync)
+            // This checks all the pre-conditions before showing NuGet manager UI at solution level
+            // without initializing full VSSolutionManager, otherwise it will create hang issues.
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var isSolutionAvailable = -1;
```

Comment:
instead of -1 , 0 and 1, either use an enum for readability or a nullable boolean.